tests/typescript/compiler/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -20,6 +20,119 @@ var results = number[];
 
 `;
 
+exports[`castOfAwait.ts 1`] = `
+// @target: es6
+async function f() {
+    <number> await 0;
+    typeof await 0;
+    void await 0;
+    await void <string> typeof <number> void await 0;
+    await await 0;
+}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// @target: es6
+async function f() {
+  <number>await 0;
+  typeof await 0;
+  void await 0;
+  await void (<string>typeof (<number>void await 0));
```

Comment:
It seems like an improvement as it's unclear how to parse it otherwise. But need to look at real typescript codebase to see what people are doing in practice